### PR TITLE
Add video cover

### PR DESCRIPTION
**Pull request by rodolphechartier**

### CODE REVIEW

File: src/webapp/src/components/VideoMiniature.vue [hunk 1]
```diff
@@ -25,13 +22,41 @@ export default {
...
+  },
   methods: {
     goToVideo: function () {
       this.$router.push({ path: `/watch/${this.id}` });
     },
+    getCover: function () {
+      axios
+        .get(
+          process.env.VUE_APP_API_ADDR +
+            `/api/v1/videos/${this.id}/cover/${this.cover.split("/")[1]}`,
```
[the remark this answers is not in the document]
Comment (rodolphechartier) [reply src/webapp/src/components/VideoMiniature.vue:45]: C'est pas faux, je peux faire cette modif

File: src/cmd/api/controllers/video_upload_handler.go [hunk 2]
```diff
@@ -45,13 +46,15 @@ type Response struct {
 // @Accept multipart/form-data
 // @Produce plain
 // @Param file formData file true "video"
+// @Param file formData file false "cover"
+// @Param file formData file true "title"
```
Comment (ClerySem) [inline src/cmd/api/controllers/video_upload_handler.go:50]: Should be : 
// @Param video formData file true "video"
// @Param cover formData file false "cover"
// @Param title formData string true "title"

File: src/cmd/api/controllers/video_upload_handler.go [hunk 3]
```diff
@@ -45,13 +46,15 @@ type Response struct {
...
 // @Success 200 {Json} Video and Links (HATEOAS)
 // @Failure 400 {object} object
 // @Failure 409 {object} object
 // @Failure 415 {object} object
 // @Failure 500 {object} object
 // @Router /api/v1/videos/upload [post]
-func (v VideoUploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (v VideoUploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) { //nolint:cyclop
```
[the remark this answers is not in the document]
Comment (RaboliotLeGris) [reply src/cmd/api/controllers/video_upload_handler.go:57]: treason

**Comment by Tezno** [comment]
@rodolphechartier Thanks for checking if the image is opensource

File: src/cmd/api/controllers/videos_list_handler.go [hunk 4]
```diff
@@ -28,62 +31,10 @@ type VideoListResponse struct {
 }
 
 type VideosListHandler struct {
+	S3Client  clients.IS3Client
```
Comment (ClerySem) [inline src/cmd/api/controllers/videos_list_handler.go:34]: It seems we don't need the S3Client here.
And then we don't need it too in the router

**Review by Isilud — DISMISSED** [review]

**Review by ClerySem — APPROVED** [review]

**Review by RaboliotLeGris — APPROVED** [review]

**Pull request merged**